app/src/main/java/com/epfl/beatlink/repository/profile/ProfileRepositoryFirestore.kt

```diff
@@ -86,6 +86,15 @@ open class ProfileRepositoryFirestore(
             // Add the username to the `usernames` collection
             val usernameDocRef = db.collection("usernames").document(profileData.username)
             transaction.set(usernameDocRef, mapOf<String, Any>())
+
+            // Add an empty document in the `friendRequests` collection for the user
+            val requestsDocRef = db.collection("friendRequests").document(userId)
+            transaction.set(
+                requestsDocRef,
+                mapOf(
+                    "ownRequests" to mapOf<String, Boolean>(),
+                    "friendRequests" to mapOf<String, Boolean>(),
+                    "allFriends" to mapOf<String, String>()))
```

Comment:
If a document in the `friendRequests` collection is created when adding a `userProfile`, shouldn't we also ensure that this document is deleted when the corresponding `userProfile` is deleted? Otherwise, we might end up with orphaned documents in Firestore.

Reply:
True ! Thank you for pointing out 👍